src/linux/agent/system-server.js

```diff
@@ -75,19 +76,29 @@ rpc.exports = {
       return result;
     });
   },
-  getProcessName: function (pkg) {
+  getProcessName: function (pkg, userid=0) {
     return performOnJavaVM(function () {
       try {
-        return packageManager.getApplicationInfo(pkg, GET_META_DATA).processName.value;
+        var UserHandle = Java.use('android.os.UserHandle');
+        var userHandle = UserHandle.of(userid);
+        return packageManager.getApplicationInfoAsUser(pkg, GET_META_DATA, userid).processName.value;
       } catch (e) {
-        throw new Error("Unable to find application with identifier '" + pkg + "'");
+        throw new Error("Unable to find application with identifier '" + pkg + "' for user " + userid);
       }
     });
   },
-  startActivity: function (pkg, activity) {
+  startActivity: function (pkg, activity, userid=0) {
     return performOnJavaVM(function () {
+
+      if(userid != 0) {
```

Comment:
```suggestion
      if (uid !== 0) {
```

(either that or `userId`)

Reply:
2d966a86f9027d314d2ecff7e38c9ff2d95ae0c7